src/pages/Processes/CreationModal/index.tsx

```diff
@@ -0,0 +1,207 @@
+import { useEffect, useState } from "react";
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  chakra,
+  Button,
+  useToast,
+  Select,
+  Text,
+  Checkbox,
+  Box,
+} from "@chakra-ui/react";
+import * as yup from "yup";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+
+import { Input } from "components/FormFields";
+import { useLoading } from "hooks/useLoading";
+import { getPriorities } from "services/priorities";
+import { getFlows } from "services/flows";
+import { createProcess } from "services/processes";
+
+type FormValues = {
+  record: string;
+  nickname: string;
+  idFlow: number;
+  idPriority: number;
+};
+
+const validationSchema = yup.object({
+  record: yup.string().required("Digite o número do Registro."),
+  nickname: yup.string().required("Dê um apelido para esse Registro."),
+  idFlow: yup.string().required("Selecione um fluxo para esse Registro"),
+  idPriority: yup.string().notRequired(),
+});
+
+interface CreationModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  afterSubmission: () => void;
+}
+
+export function CreationModal({
+  isOpen,
+  onClose,
+  afterSubmission,
+}: CreationModalProps) {
+  const toast = useToast();
+  const { handleLoading } = useLoading();
+  const [legalPriority, setLegalPriority] = useState(false);
+  const [priorities, setPriorities] = useState<Priority[]>();
+  const [flows, setFlows] = useState<Flow[]>();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<FormValues>({
+    resolver: yupResolver(validationSchema),
+    reValidateMode: "onChange",
+  });
+
+  const onSubmit = handleSubmit(async (formData) => {
+    console.log("ENTREI SUBMIT");
+    handleLoading(true);
+    const body = {
+      record: formData.record,
+      nickname: formData.nickname,
+      idFlow: formData.idFlow,
+      priority: legalPriority ? formData.idPriority : 0,
+      effectiveDate: new Date(),
+    };
+
+    const res = await createProcess(body);
+
+    onClose();
+    afterSubmission();
+
+    if (res.type === "success") {
+      handleLoading(false);
+
+      toast({
+        id: "create-process-success",
+        title: "Sucesso!",
+        description: "O Processo foi criado.",
+        status: "success",
+      });
+      return;
+    }
+
+    handleLoading(false);
+    toast({
+      id: "create-process-error",
+      title: "Erro ao criar processo",
+      description: res.error?.message,
+      status: "error",
+      isClosable: true,
+    });
+  });
+
+  const handleGetPriorities = async () => {
+    const res = (await getPriorities()).value;
+    setPriorities(res);
+  };
+
+  const handleGetFlows = async () => {
+    const res = (await getFlows()).value;
+    setFlows(res);
+  };
+
+  useEffect(() => {
+    handleGetFlows();
```

Comment:
Repete um comentário que mandei mais abaixo, só que to revisando de baixo pra cima entao vou colar aqui de novo: 

Aqui ao invés de usar o useEffect acredito que valha seguir o padrão que já tava antes, usando o useQuery, teve algum problema com isso aqui? Se tiver só falar que a gente tenta resolver